nixos/modules/services/web-apps/pds.nix

```
@@ -0,0 +1,232 @@
+{
+  options.services.pds = {
+    enable = mkEnableOption "pds";
+
+    package = mkPackageOption pkgs "pds" { };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = types.attrsOf (types.nullOr types.str);
+        options = {
+          PDS_PORT = mkOption {
+            type = types.str;
+            default = "3000";
+            description = "Port to listen on";
+          };
+
+          PDS_HOSTNAME = mkOption {
+            type = types.str;
+            example = "pds.example.com";
+            description = "Instance hostname (base domain name)";
+          };
+
+          PDS_BLOB_UPLOAD_LIMIT = mkOption {
+            type = types.str;
+            default = "52428800";
+            description = "Size limit of uploaded blobs";
+          };
+
+          PDS_DID_PLC_URL = mkOption {
+            type = types.str;
+            default = "https://plc.directory";
+            description = "URL of DID PLC directory";
+          };
+
+          PDS_BSKY_APP_VIEW_URL = mkOption {
+            type = types.str;
+            default = "https://api.bsky.app";
+            description = "URL of bsky frontend";
+          };
+
+          PDS_BSKY_APP_VIEW_DID = mkOption {
+            type = types.str;
+            default = "did:web:api.bsky.app";
+            description = "DID of bsky frontend";
+          };
+
+          PDS_REPORT_SERVICE_URL = mkOption {
+            type = types.str;
+            default = "https://mod.bsky.app";
+            description = "URL of mod service";
+          };
+
+          PDS_REPORT_SERVICE_DID = mkOption {
+            type = types.str;
+            default = "did:plc:ar7c4by46qjdydhdevvrndac";
+            description = "DID of mod service";
+          };
+
+          PDS_CRAWLERS = mkOption {
+            type = types.str;
+            default = "https://bsky.network";
+            description = "URL of crawlers";
+          };
+
+          PDS_DATA_DIRECTORY = mkOption {
+            type = types.str;
+            default = "/var/lib/pds";
+            description = "Directory to store state";
+          };
+
+          PDS_BLOBSTORE_DISK_LOCATION = mkOption {
+            type = types.nullOr types.str;
+            default = "/var/lib/pds/blocks";
+            description = "Store blobs at this location, set to null to use e.g. S3";
+          };
+
+          LOG_ENABLED = mkOption {
+            type = types.nullOr types.str;
+            default = "true";
+            description = "Enable logging";
+          };
+        };
+      };
+
+      description = ''
+        Environment variables to set for the service. Secrets should be
+        specified using {option}`environmentFile`.
+
+        Refer to <https://github.com/bluesky-social/atproto/blob/92cd7a84ad207278c241afce8c8491e73b0a24e0/packages/pds/src/config/env.ts> for available environment variables.
```

```suggestion
        Refer to <https://github.com/bluesky-social/atproto/blob/main/packages/pds/src/config/env.ts> for available environment variables.
```
or we link at the exact version of the package. Hardcoding some commit here is a bad idea